src/Finch_SolidificationData.hpp

```diff
@@ -267,72 +267,98 @@ class SolidificationData
         fout.close();
     }
 
-    Kokkos::Array<double, 6> getBounds()
+    std::array<double, 3> getLowerBounds()
     {
         // Local copies for lambda capture
         auto events_ = events;
         auto count_host =
             Kokkos::create_mirror_view_and_copy( Kokkos::HostSpace(), count );
 
-        // Iterate over list of events, getting the min/max bounds in each
-        // direction
-        double x_min, y_min, z_min, x_max, y_max, z_max;
+        // Iterate over list of events, getting the min bounds in each direction
+        double x_min, y_min, z_min;
         Kokkos::parallel_reduce(
             "solidification_event_bounds", count_host( 0 ),
             KOKKOS_LAMBDA( const int& n, double& x_min_th, double& y_min_th,
-                           double& z_min_th, double& x_max_th, double& y_max_th,
-                           double& z_max_th ) {
+                           double& z_min_th ) {
                 double x_event = events_( n, 0 );
                 double y_event = events_( n, 1 );
                 double z_event = events_( n, 2 );
                 if ( x_event < x_min_th )
                     x_min_th = x_event;
-                if ( x_event > x_max_th )
-                    x_max_th = x_event;
                 if ( y_event < y_min_th )
                     y_min_th = y_event;
-                if ( y_event > y_max_th )
-                    y_max_th = y_event;
                 if ( z_event < z_min_th )
                     z_min_th = z_event;
+            },
+            Kokkos::Min<double>( x_min ), Kokkos::Min<double>( y_min ),
+            Kokkos::Min<double>( z_min ) );
+
+        // Get the min bounds on each direction across all ranks
+        std::array<double, 3> finch_data_bounds_low;
```

Comment:
Specifying that this is a finch bound seems unncessary

Reply:
Fair enough, fixed